src/data/fin.lean

```diff
@@ -228,14 +267,18 @@ end
 
 section rec
 
+/-- Define `C n i` by induction on  `i : fin n` interpreted as `(0 : fin (n - i)).succ.succ…`,
+i.e., `C n i = Hs $ … Hs H0)` -/
 @[elab_as_eliminator] def succ_rec
   {C : ∀ n, fin n → Sort*}
   (H0 : ∀ n, C (succ n) 0)
-  (Hs : ∀ n i, C n i → C (succ n) i.succ) : ∀ {n : ℕ} (i : fin n), C n i
+  (Hs : ∀ n i, C n i → C (succ n) i.succ) : Π {n : ℕ} (i : fin n), C n i
 | 0        i           := i.elim0
 | (succ n) ⟨0, _⟩      := H0 _
 | (succ n) ⟨succ i, h⟩ := Hs _ _ (succ_rec ⟨i, lt_of_succ_lt_succ h⟩)
 
+/-- Define `C n i` by induction on  `i : fin n` interpreted as `(0 : fin (n - i)).succ.succ…`,
```

Comment:
This is a little confusing. For one, `Hs` isn't introduced in the doc string. It's not immediately clear what the `...` is doing. Instead of the short one-line description, could you maybe expand it to a few sentences that make it clearer how to use the definition?

Reply:
Reformulated. Is it better?